rfc-079-script-imports-of-production-data.md

```diff
@@ -0,0 +1,56 @@
+# Script imports of production data
+
+## Summary
+
+Developers writing production data to file on a local or remote server MUST do
+this via an automated script, rather than doing this manually, e.g. via the
+Rails console.
+
+This script MUST anonymise any sensitive data, and the data dump MUST be stored
+outside of the working repository to ensure it can't be inadvertedly committed
+alongside any changes to the codebase.
+
+## Problem
+
+Fetching data from a database by hand is error-prone. A data dump may be left
+on a server, committed to a repository or even in some cases alter production
+data in unforeseen ways. Developers may not realise they have exposed or
+impacted production data until long after an incident occurs.
+
+By writing a script to automate this process, we ensure that every import of
+data from a production environment is done in the same way, minimising room for
+error. It also allows us to scrub the resulting data dump of any
+personally-identifying and security-sensitive information.
+
+Committing a data-fetching script to the repository makes it visible to other
+members of the team in the code review process, and allows others to provide
+feedback to ensure we're reaching a consensus on the safest way of fetching
+this data.
+
+## Proposal
+
+If developing an application requres a developer to write any production data
+containing sensitive data to a local file or a file on a remove server, this
+MUST be done via an automated script.
+
+The script MUST anonymise any personally-identifying or security-sensitive
+data.
+
+The automated script MUST have gone through the usual code review process with
+other members of the team.
+
+The file written to by the script MUST be stored outside the repository or in a
```

Comment:
It should be written outside _any_ Git repository. Probably worth stating that as a separate point / sentence to really drive the point home.

Reply:
Are you suggesting this as an additional SHOULD e.g. It MUST be stored outside the repository && SHOULD be stored outside any Git repository, or does "It MUST be stored outside any Git repository" cover both? (I guess the emphasis on MUST/SHOULD is what I'm asking here).

Reply:
is `/tmp`, `$TMPDIR` or equivalent the right place to store this? Perhaps "it MUST be stored the system's temporary file storage location".

Reply:
I believe all the major unixes (including macOS) automatically remove things in `/tmp` after a period of time. macOS has it at 3 days according to https://superuser.com/a/187105, and many others are described in https://serverfault.com/a/377349.

Reply:
Yup! And I'm suggesting an additional MUST, either rolled together into a single MUST, or as two separate MUSTs for emphasis.

Reply:
But yes, what James said.

Reply:
Thanks both, have added some info about this.